### PR TITLE
untappd support

### PR DESCRIPTION
**Pull request by TerribleDev**
This should in theory provide untappd support. However I have yet to test this, as I waiting on an API key from the untappd folks, which can take upward of 2-3 weeks. I believe it is being maintained by two people part time.

I am adding this here, so atleast people can start looking at this, but I wouldn't suggest merging it in until I can fully test this.

If someone else has an Untappd token, then I would suggest that person start really checking this over.

### CODE REVIEW

**Comment by AlexArchive** [comment]
Many thanks for your contribution, @tparnell8.

I will have a proper look at this once you have tested it. In the mean time, I will apply for my own API key so I can test this too :bear:.

**Comment by jerriep** [comment]
Hi @tparnell8, any news yet on that API key?

**Comment by TerribleDev** [comment]
I just got it...and this is not quite working, looking into it now.

**Comment by jerriep** [comment]
Awesome :) I am pushing to clear the PR backlog so we can do a release over the weekend or early next week. 

No worries though if you don't have time to get it sorted now. We can do another release at a later stage with the Untappd support

**Comment by TerribleDev** [comment]
@jerriep thats fine, if you wish to close this for now, its ok by me..I can always submit a new one

**Comment by jerriep** [comment]
No need to close it :) When you get around to it just add changes to this PR.

**Comment by rodkings** [comment]
Wow, I was just about to start the Untappd provider, decided to check here and here it is.
@tparnell8 if you need any help testing, please let me know. I'm still waiting for my key.

**Comment by TerribleDev** [comment]
@rodkings I have ran out of time to finish this, so if you could do the things I would certainly appreciate it.

**Comment by rodkings** [comment]
@tparnell8 Oh right man, I will continue the work then. Thank you.

**Comment by rodkings** [comment]
What is the best way to continue your work? I'm kinda new to github.

**Comment by rodkings** [comment]
Ops, it's all ok. I was able to clone my fork and add the upstream from here and checkout this pull request.

@tparnell8 So, Were you getting 404 on the callback url? /signin-Untappd=?code=... is giving me 404.

**Comment by TerribleDev** [comment]
@rodkings that sounds correct...

**Comment by rodkings** [comment]
@tparnell8 I was able to fix adding the State to cookie, had no state saving at all. I just need to figure out how to commit to your pull request =/

**Comment by TerribleDev** [comment]
@rodkings submit a pull request against my fork of OWINOAuthProvider master branch https://github.com/tparnell8/OwinOAuthProviders once I merge it into my master, it will update this pull request and add your commits.

**Comment by TerribleDev** [comment]
@rodkings  if you can't figure it out, just push the code to your fork, and let me know

**Comment by rodkings** [comment]
@tparnell8 Done.

**Comment by TerribleDev** [comment]
@rodkings @jerriep @ByteBlast this appears to be all set at this time. Untappd OAuth is now a thing, with an extra contrib by @rodkings we got this done!

File: OwinOAuthProvidersDemo/App_Start/Startup.Auth.cs [hunk 1]
```diff
@@ -28,8 +28,7 @@
 using Owin.Security.Providers.OpenID;
 using Owin.Security.Providers.SoundCloud;
 using Owin.Security.Providers.Steam;
-using Owin.Security.Providers.Wargaming;
-using Owin.Security.Providers.WordPress;
+using Owin.Security.Providers.Wargaming;using Owin.Security.Providers.Untappd;using Owin.Security.Providers.WordPress;
```
Comment (TerribleDev) [inline OwinOAuthProvidersDemo/App_Start/Startup.Auth.cs:31]: my mistake for all the whitespace commits in this file.

Comment (rodkings) [inline OwinOAuthProvidersDemo/App_Start/Startup.Auth.cs:31]: thank you @tparnell8

**Comment by TerribleDev** [comment]
@jerriep @ByteBlast When can we expect this to be merged in?

**Comment by jerriep** [comment]
I'll try and get around to it this weekend guys, sorry...

File: Owin.Security.Providers/Untappd/UntappdAuthenticationOptions.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        public ISecureDataFormat<AuthenticationProperties> StateDataFormat { get; set; }
+
+        /// <summary>
+        ///     Initializes a new <see cref="UntappdAuthenticationOptions" />
+        /// </summary>
+        public UntappdAuthenticationOptions()
+            : base("Untappd")
+        {
+            Caption = Constants.DefaultAuthenticationType;
+            CallbackPath = new PathString("/signin-Untappd");
```
Comment (jerriep) [inline Owin.Security.Providers/Untappd/UntappdAuthenticationOptions.cs]: Please make everything lowercase in this string, i.e. signin-untappd

**Comment by TerribleDev** [comment]
@jerriep  done

**Comment by jerriep** [comment]
@tparnell8 We are looking for help moving to ASP.NET 5. Please read http://www.jerriepelser.com/blog/calling-contributors-for-oauth-providers

**Comment by TerribleDev** [comment]
Sure...where are we putting the new code?

**Comment by jerriep** [comment]
https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers

**Comment by jerriep** [comment]
@tparnell8 Please also just read the blog post to take not of the things I mentioned in there

**Pull request merged**